contrib/supervisord-monitor.php

```diff
@@ -0,0 +1,153 @@
+    'process_name' => 'process01',
+]);
+```
+
+or
+
+```
+set('supervisord_uri', 'https://youruri.xyz/supervisor');
+set('supervisord_basic_auth_user', 'username');
+set('supervisord_basic_auth_password', 'password');
+set('supervisord_process_name', 'process01');
+```
+
+- `supervisord` – array with configuration for Supervisord
+    - `uri` – URI to the Supervisord monitor page
+    - `basic_auth_user` – Basic auth username to access the URI
+    - `basic_auth_password` – Basic auth password to access the URI
+    - `process_name` – the process name, as visible in the Supervisord monitor page. Multiple processes can be listed here, comma separated
+
+### Task
+
+- `supervisord-monitor:restart` Restarts given processes
+
+### Usage
+
+A complete example with configs, staging and deployment
+
+```
+<?php
+
+namespace Deployer;
+use Dotenv\Dotenv;
+
+require 'vendor/autoload.php';
+
+require 'supervisord_monitor.php';
+
+// Project name
+set('application', 'myproject.com');
+
+// Project repository
+set('repository', 'git@github.com:myorg/myproject.com');
+
+set('supervisord', [
+    'uri' => 'https://youruri.xyz/supervisor',
+    'basic_auth_user' => 'username',
+    'basic_auth_password' => 'password',
+    'process_name' => 'process01',
+]);
+
+host('staging.myproject.com')
+    ->set('branch', 'develop')
+    ->set('labels' => ['stage' => 'staging']);
+
+host('myproject.com')
+    ->set('branch', 'main')
+    ->set('labels' => ['stage' => 'production']);
+
+// Tasks
+task('build', function () {
+    run('cd {{release_path}} && build');
+});
+
+task('deploy', [
+    'build',
+    'supervisord',
+])
+
+task('supervisord', ['supervisord-monitor:restart'])
+    ->select('stage=production');
+```
+ */
+
+namespace Deployer;
+
+use Deployer\Utility\Httpie;
+
+function checkConfig()
+{
+    $config = get('supervisord', []);
+    foreach ($config as $key => $value) {
+        set('supervisord_' . $key, $value);
+    }
+
+    if (!get('supervisord_uri') ||
+        !get('supervisord_basic_auth_user') ||
+        !get('supervisord_basic_auth_password') ||
+        !get('supervisord_process_name')) {
+        throw new \RuntimeException("<comment>Please configure Supervisord config:</comment> <info>set('supervisord', array('uri' => 'yourdomain.xyz/supervisor', 'basic_auth_user' => 'abc' , 'basic_auth_password' => 'xyz', 'process_name' => 'process01,process02'));</info> or <info>set('supervisord_uri', 'yourdomain.xyz/supervisor'); set('supervisord_basic_auth_user', 'abc'); etc</info>");
+    }
+}
+
+function getBasicAuthToken()
+{
+    return 'Basic ' . base64_encode(get('supervisord_basic_auth_user'). ':'. get('supervisord_basic_auth_password'));
+}
+
+function isAuthenticated()
+{
+    checkConfig();
+
+    $authResponseInfo = [];
+    Httpie::post(get('supervisord_uri'))->header('Authorization', getBasicAuthToken())->send($authResponseInfo);
+
+    return $authResponseInfo['http_code'] === 200;
+}
+
+function action($name, $action = 'stop')
```

Comment:
This very broad name in Deployer namespace. 

Please refactor all function to use as variables via set(name, function).

Reply:
I renamed the function names to be preficed with the recipe name.

What do you mean by refactoring all functions to use as variables via set(name, function)
There should only be one action available: ```supervisord-monitor:restart```

Reply:
@antonmedv updated the code with one last change so that stop and start are also available tasks and also now wrapped so that it won't interfere with the other code inside the Deployer namespace